TypeSystem/variant.h

```diff
@@ -239,7 +265,7 @@ struct VariantImpl<NAME, TypeListImpl<TYPES...>> : CurrentVariantNameHelper<NAME
   }
 
   template <typename X>
-  std::enable_if_t<!std::is_same<X, CurrentStruct>::value, X&> VariantValueImpl() {
+  std::enable_if_t<!std::is_same<X, current::variant::object_base_t>::value, X&> VariantValueImpl() {
```

Comment:
Hmm. How are these `enable_if`-s for `VariantValueImpl` & `VariantExistsImpl` are supposed to work?